hera_qm/firstcal_metrics.py

```diff
@@ -384,30 +388,23 @@ def __init__(self, calfits_files, use_gp=True):
             # noise_level=0.01 are just initial conditions and are not the final hyperparameter solution
             kernel = gp.kernels.RBF(length_scale=0.2, length_scale_bounds=(0.01, 1.0)) + gp.kernels.WhiteKernel(noise_level=0.01)
             x = self.frac_JD.reshape(-1, 1)
-            self.delay_smooths = []
+            self.delay_smooths = copy.copy(self.delay_fluctuations)
             # iterate over each antenna
             for i in range(self.Nants):
                 # get ydata
                 y = copy.copy(self.delay_fluctuations[i, :, :])
                 # scale by std
-                ystd = np.sqrt(astats.biweight_midvariance(y, axis=0))
-                # if this is a simulation it is possible all the delays are 1
-                # In such a case the array ystd will all be zeros
-                if all(_ystd == 0 for _ystd in ystd.ravel()):
-                    self.delay_smooths.append(self.delays[i, :, :])
-                else:
-                    y /= ystd
-                    GP = gp.GaussianProcessRegressor(kernel=kernel, n_restarts_optimizer=0)
-                    ymodel_stack = np.zeros((self.Ntimes, self.Npols))
-                    for pol_cnt in range(self.Npols):
-                        # fit GP and remove from delay fluctuations
-                        # but only one polarization at a time
+                ystd = np.sqrt([astats.biweight_midvariance(y[~delay_flags[i, :, ip], ip]) for ip in range(self.Npols)])
+                y /= ystd
+                GP = gp.GaussianProcessRegressor(kernel=kernel, n_restarts_optimizer=0)
+                ymodel_stack = np.zeros((self.Ntimes, self.Npols))
```

Comment:
It seems like this variable is no longer used. Can we just remove it?

Reply:
You're right. I removed it.